src/main/java/org/jenkinsci/plugins/workflow/support/pickles/serialization/PickleResolver.java

```diff
@@ -49,15 +51,28 @@
      */
     private List<Object> values;
 
+    private final FlowExecutionOwner owner;
+
+    @Deprecated
     public PickleResolver(List<? extends Pickle> pickles) {
+        this(pickles, FlowExecutionOwner.dummyOwner());
+    }
+
+    public PickleResolver(List<? extends Pickle> pickles, FlowExecutionOwner owner) {
         this.pickles = pickles;
+        this.owner = owner;
     }
 
     public Object get(int id) {
         return values.get(id);
     }
 
+    @Deprecated
     public ListenableFuture<PickleResolver> rehydrate() {
+        return rehydrate(new ArrayList<ListenableFuture<?>>());
```

Comment:
Why don't you just use the same implementation in https://github.com/jenkinsci/workflow-api-plugin/pull/5?